## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#82ac7fd5-5508-4536-91c1-ba2023a81662) |
| mint-fs.sh | :x: [more...](#28601003-018c-43f4-9517-40e3231a7f6c) |
| mint-dist-xl.sh | :x: [more...](#340f48f5-17ec-4b05-86f3-55a1abfaa115) |
<a name="82ac7fd5-5508-4536-91c1-ba2023a81662"></a>
### 8314-32a9113/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.51:32305
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 53a79948cd52:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 2 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 51 seconds
(4/14) Running aws-sdk-ruby tests ... done in 15 seconds
(5/14) Running awscli tests ... done in 2 minutes and 41 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 1 minutes and 22 seconds
(8/14) Running minio-dotnet tests ... done in 1 minutes and 42 seconds
(9/14) Running minio-go tests ... FAILED in 4 minutes and 16 seconds
{
  "args": {},
  "duration": 114098,
  "error": "Please reduce your request",
  "function": "testStorageClassMetadataCopyObject()",
  "message": "CopyObject failed on RRS",
  "name": "minio-go: testStorageClassMetadataCopyObject",
  "status": "FAIL"
}

Executed 8 out of 14 tests successfully.
```
<a name="28601003-018c-43f4-9517-40e3231a7f6c"></a>
### 8314-32a9113/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.54:32632
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp ee5a5d1a6475:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 41 seconds
(4/14) Running aws-sdk-ruby tests ... done in 2 seconds
(5/14) Running awscli tests ... done in 1 minutes and 52 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 44 seconds
(8/14) Running minio-dotnet tests ... done in 31 seconds
(9/14) Running minio-go tests ... done in 58 seconds
(10/14) Running minio-java tests ... done in 51 seconds
(11/14) Running minio-js tests ... done in 46 seconds
(12/14) Running minio-py tests ... FAILED in 2 minutes and 10 seconds
{
  "args": {
    "bucket_name": "minio-py-test-15693ca7-26d7-4cb2-a1b4-66323d175bf5",
    "response._body": "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>AccessDenied</Code><Message>Request has expired</Message><Key>c4097e99-da72-4263-b07f-bdfbd89dc455</Key><BucketName>minio-py-test-15693ca7-26d7-4cb2-a1b4-66323d175bf5</BucketName><Resource>/minio-py-test-15693ca7-26d7-4cb2-a1b4-66323d175bf5/c4097e99-da72-4263-b07f-bdfbd89dc455</Resource><RequestId>15C7DC07D76AFE3F</RequestId><HostId>16e61715-f20d-4d34-9854-f43db8b334d5</HostId></Error>",
    "response.headers": "{\"_container\": {\"accept-ranges\": [\"Accept-Ranges\", \"bytes\"], \"content-length\": [\"Content-Length\", \"440\"], \"content-security-policy\": [\"Content-Security-Policy\", \"block-all-mixed-content\"], \"content-type\": [\"Content-Type\", \"application/xml\"], \"server\": [\"Server\", \"MinIO/DEVELOPMENT.2019-09-26T02-07-08Z\"], \"vary\": [\"Vary\", \"Origin\"], \"x-amz-request-id\": [\"X-Amz-Request-Id\", \"15C7DC07D76AFE3F\"], \"x-xss-protection\": [\"X-Xss-Protection\", \"1; mode=block\"], \"date\": [\"Date\", \"Thu, 26 Sep 2019 02:28:41 GMT\"]}}",
    "object_name": "c4097e99-da72-4263-b07f-bdfbd89dc455",
    "presigned_get_object_url": "http://72.28.97.54:32632/minio-py-test-15693ca7-26d7-4cb2-a1b4-66323d175bf5/c4097e99-da72-4263-b07f-bdfbd89dc455?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minio%2F20190926%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20190926T022834Z&X-Amz-Expires=5&X-Amz-SignedHeaders=host&X-Amz-Signature=6a8e6ad7d856dbd0b6271623471d872854f5088d087e07cf4ffef5d212037486",
    "response.status": 403,
    "response.reason": "Forbidden"
  },
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 1341, in test_presigned_get_object_expiry_5sec\n    object_name).get_exception()\nminio.error.AccessDenied: AccessDenied: message: Access Denied\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 2036, in main\n    test_presigned_get_object_expiry_5sec(client, log_output)\n  File \"/mint/run/core/minio-py/tests.py\", line 1355, in test_presigned_get_object_expiry_5sec\n    raise Exception(err)\nException: AccessDenied: message: Access Denied\n",
  "message": "AccessDenied: message: Access Denied",
  "duration": 54,
  "status": "FAIL",
  "name": "minio-py:test_presigned_get_object_expiry_5sec",
  "function": "presigned_get_object(bucket_name, object_name, expires, response_headers, request_date)"
}

Executed 11 out of 14 tests successfully.
```
<a name="340f48f5-17ec-4b05-86f3-55a1abfaa115"></a>
### 8314-32a9113/mint-dist-xl.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.59:31293
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 789c8727bccb:/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 0 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 2 minutes and 6 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... FAILED in 19 seconds
{
  "name": "mc",
  "duration": "7848",
  "function": "test_get_object_multipart",
  "status": "FAIL",
  "error": "/mint/run/core/mc/mc --config-dir /tmp/.mc-27943 --quiet --no-color cp myminio/mc-test-bucket-21167/mc-test-object-27372 mc-test-object-27372.downloaded  >>> \n`myminio/mc-test-bucket-21167/mc-test-object-27372` -> `mc-test-object-27372.downloaded`\nmc: <ERROR> Failed to copy `http://72.28.97.59:31293/mc-test-bucket-21167/mc-test-object-27372`. read tcp 172.17.0.10:48634->72.28.97.59:31293: read: connection reset by peer\nmc: <ERROR> Session safely terminated. To resume session `mc session resume XDkDAlSk`"
}

Executed 6 out of 14 tests successfully.
```

--- a/cmd/http-tracer.go
+++ b/cmd/http-tracer.go
@@ -100,8 +100,10 @@ func Trace(f http.HandlerFunc, logBody bool, w http.ResponseWriter, r *http.Requ
 
 	// Setup a http request body recorder
 	reqHeaders := r.Header.Clone()
-	reqHeaders.Set("Content-Length", strconv.Itoa(int(r.ContentLength)))
 	reqHeaders.Set("Host", r.Host)
+	if len(r.TransferEncoding) == 0 {
+		reqHeaders.Set("Content-Length", strconv.Itoa(int(r.ContentLength)))
+	}
 	for _, enc := range r.TransferEncoding {
 		reqHeaders.Add("Transfer-Encoding", enc)
 	}

